### DISCUSSION TITLE
Lathe problem

### DISCUSSION THREAD

**Discussion by jal123sin** [terjeio/ioSender#300] [Q&A] [answered]
Hello, the iosender lathe radius mode and diameter mode problem, it seems that it is impossible to distinguish the gcode x value is set as diameter of the cylinder.  As the sample cut depth I set is 1mm, it show 2mm cut depth on the main display area. I have tried to switch between the radius and diameter mode on tag of Settings: App, it can only displayed the 2mm cut path. Please advise a way to solve this problem.

**Comment by terjeio** [comment]
Is lathe mode enabled in the controller? Settings: _Grbl > General > Mode of operation_
G7/G8 is used to switch between diameter/radius in the controller, current mode is displayed above the X DRO:

![image](https://github.com/terjeio/ioSender/assets/20260062/7dddcdbe-1d28-45cf-b0a2-0683d69462eb)

_Settings:App_ lathe settings are used by the lathe wizards.

**Reply by jal123sin** [reply]
Thanks for your promptly reply, I try to change the radius mode to diameter on the tag (Setting: App) as figure below but the DRO section still display Radius.
![pic1](https://github.com/terjeio/ioSender/assets/136598254/0e15674d-48f2-4067-9ad0-b193730df4ce)

**Reply by terjeio** [reply]
Send G7 from the MDI or add it to your gcode file to change to diameter mode in the controller. Settings: App configuration is for the wizards.

**Comment by jal123sin** [comment]
Thanks, it great I add the G7 into the nc file, it's working. But I just find another problem on the G3 r value seems not functioning. I attached the nc file.
![pic2](https://github.com/terjeio/ioSender/assets/136598254/cf42ae08-62ec-4d12-af99-74e3f375d2d8)
[gru_test2.txt](https://github.com/terjeio/ioSender/files/11756573/gru_test2.txt)

**Reply by terjeio** [reply]
Did you change the plane to XZ (G18)?
ioSender may display arcs as straight lines depending on settings, this to speed up rendering:

![image](https://github.com/terjeio/ioSender/assets/20260062/cb9fc3d2-7fee-46e8-8689-946acf70f16b)

**Reply by jal123sin** [reply]
I just add G18 in front of line G3 and I didn't change the Arc resolution, the former value is (10) . Now I try 40 segments but no change, still straight line display.

**Reply by terjeio** [reply]
G18 changes the plane of the arc, which has to be XZ if I am not mistaken. The resolution of the display of the arc is controlled by both _Arc resolution_ and _Min distance_, reduce min distance to get more detail. These settings has nothing to do with how the arc is executed by the controller so it is not very important.

**Reply by jal123sin** [reply, marked as answer]
Ok, G18 works. It's displaying round corners now.
![pic3](https://github.com/terjeio/ioSender/assets/136598254/b8919d60-4d38-4b5d-aa13-73d3a8e71ea3)
